database/migrations/2021_12_12_174255_create_order_products_table.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateOrderProductsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('order_products', function (Blueprint $table) {
+            $table->id();
+            $table->foreignId('order_id')
+                ->references('id')->on('orders')
+                ->cascadeOnUpdate()
+                ->cascadeOnDelete();
+            $table->foreignId('product_id')->nullable(true)
```

Comment:
why it should be nullable?

Reply:
To be able to delete a product from database without needing to delete all information related to order_products or order_product_toppings.
Example: if we delete an order_product_id from order_products table because product_id got deleted, order_product_id gets NULL value, although we still have order_product_topping_id related to that order_product_id.
On the other hand if we use restrictOnDelete we won't be able to delete any products thus making too much redundancy in our products table.

Reply:
In general it is not desired to remove products from the database if they are used in the order. So restrict and not null is better in this context. To avoid listing of this products often are used the flag approach, like active column.